lib/_stream_writable.js

```diff
@@ -760,14 +751,6 @@ function onCorkedFinish(corkReq, state, err) {
 }
 
 function onFinished(stream, state, cb) {
-  if (state.destroyed && state.errorEmitted) {
```

Comment:
@lundibundi: The `errored` check discussed in the other section is a cleaner variant of this.